swagger.yaml

```diff
@@ -146,226 +328,821 @@ paths:
+        400:
+          description: "instance_id does not match any instances"
+        404:
+          description: "dimension_id does not match any dimensions within the instance"
+        500:
+          $ref: '#/responses/InternalError'
+    put:
+      tags:
+      - "Private"
+      summary: "Update the number of observations within this instance"
+      description: "Update the number of observations within this instance"
+      parameters:
+      - $ref: '#/parameters/instance_id'
+      - $ref: '#/parameters/instance'
+      security:
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "The event was added to the instance"
+        400:
+          description: "Invalid json message sent"
+        401:
+          $ref: '#/responses/UnauthorisedError'
+        403:
+          $ref: '#/responses/ForbiddenError'
+        404:
+          description: "InstanceId does not match any instances"
+        500:
+          $ref: '#/responses/InternalError'
+  /instances/{instance_id}/events:
+    post:
+      tags:
+       - "Private"
+      summary: "Add an event to an instance"
+      description: |
+        Add a new event into an instance. Events can be for information or error messages.
+        Each event must contain a type of event (Info or Error), a message to explain
+        the event, time of the event and finally the message offset in kafka
+      parameters:
+      - $ref: '#/parameters/instance_id'
+      - $ref: '#/parameters/event'
+      security:
+      - InternalAPIKey: []
+      responses:
+        201:
+          description: "The event was added to the instance"
+        400:
+          description: "Invalid json message sent"
+        401:
+          $ref: '#/responses/UnauthorisedError'
+        403:
+          $ref: '#/responses/ForbiddenError'
+        404:
+          description: "InstanceId does not match any instances"
+        500:
+          $ref: '#/responses/InternalError'
+  /instances/{instance_id}/inserted_observations/{inserted_observations}:
+    put:
+      tags:
+      - "Private"
+      summary: "Increment the observation inserted count"
+      description: "This will add to the number already store in the api"
+      parameters:
+      - $ref: '#/parameters/instance_id'
+      - $ref: '#/parameters/inserted_observations'
+      security:
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "Added value to inserted observation"
+        400:
+          description: "Invalid value provided for inserted_observations"
+        401:
+          $ref: '#/responses/UnauthorisedError'
+        403:
+          $ref: '#/responses/ForbiddenError'
+        404:
+          description: "InstanceId does not match any instances"
+        500:
+          $ref: '#/responses/InternalError'
+
+  /instances/{instance_id}/dimensions/{dimension}/options/{value}:
+    put:
+      tags:
+      - "Private"
+      summary: "Add a dimension for the instance"
```

Comment:
Consider making this specific to an option "Add an option to a dimension within the instance"

Reply:
done